main.cpp

```diff
@@ -425,6 +539,35 @@ IntType& IntType::divide(int rhs)
     return *this;
 }
 
+IntType& IntType::pow(const IntType& exp)
+{
+    powInternal(int(exp));
+    return *this;
+}
+
+IntType& IntType::pow(const DoubleType& exp)
+{
+    powInternal(static_cast<int>(double(exp)));
+    return *this;
+}
+
+IntType& IntType::pow(const FloatType& exp)
+{
+    powInternal(static_cast<int>(float(exp)));
+    return *this;
+}
+
+IntType& IntType::pow(int exp)
+{
+    powInternal(exp);
+    return*this;
+}
```

Comment:
after you make `powInternal` chainable, revise these functions to return what `powInternal` returns.
- [ ] revise